packages/flutter_tools/lib/src/android/deferred_components_prebuild_validator.dart

```diff
@@ -225,9 +225,15 @@ class _DeferredComponentAndroidFiles {
   Directory get componentDir => androidDir.childDirectory(name);
 
   File get androidManifestFile => componentDir.childDirectory('src').childDirectory('main').childFile('AndroidManifest.xml');
-  File get buildGradleFile => componentDir.childFile('build.gradle');
+  File get buildGradleFile {
+    if (componentDir.childFile('build.gradle.kts').existsSync()) {
```

Comment:
Looks like this logic did not get modified to the new behavior. 
This is preferring the kotlin file over the groovy one. 

Blocking